### ISSUE TITLE
[java] A false negative about the rule UnsynchronizedStaticFormatter

### ISSUE THREAD

**Issue by LynnBroe** [pmd/pmd#4688]
**Affects PMD Version: 7.0.0-rc3** 

**Rule:UnsynchronizedStaticFormatter**
<https://docs.pmd-code.org/latest/pmd_rules_java_multithreading.html#unsynchronizedstaticformatter>

**Description:**
When I call `format()` in a method, PMD will report a violation if the method does not have both a `synchronized` and `static` modifier.
That's right. However, if I use an annotation along with StaticFormatter, PMD will not report a violation and this is a false negative.

**Code Sample demonstrating the issue:**

```java
import java.text.SimpleDateFormat;
import java.util.Date;
public class Foo2 {
   @MyAnnotation
   private static final SimpleDateFormat sdf = new SimpleDateFormat();
   synchronized void bar() { // should be static, but PMD does not report
       sdf.format(new Date());
   }
}
```

**Expected outcome:**

PMD does not report a violation at line 6, but that's wrong. It is a false negative.

**Running PMD through:** *CLI*
